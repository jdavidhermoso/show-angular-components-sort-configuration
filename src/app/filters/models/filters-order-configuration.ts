export interface FiltersOrderConfiguration {
  price: number;
  language: number;
  rating: number;
  duration: number;
  difficultLevel: number;
}
