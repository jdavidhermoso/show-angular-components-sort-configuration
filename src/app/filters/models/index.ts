export * from './filters-order-configuration';
