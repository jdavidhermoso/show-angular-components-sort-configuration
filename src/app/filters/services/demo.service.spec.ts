import {DemoService} from './demo.service';
import {FiltersOrderConfiguration} from '../models';
import {BehaviorSubject} from 'rxjs';
import {skip} from 'rxjs/operators';

describe('FiltersService', () => {
  let service: DemoService;

  beforeEach(() => {
    service = new DemoService();

  });

  it('Should notify when a new value is emitted', (done) => {
    service.filtersOrderConfigurationBehaviorSubject = new BehaviorSubject<FiltersOrderConfiguration>({
      price: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      difficultLevel: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      language: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      rating: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      duration: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION
    });

    service.filtersOrderConfigurationBehaviorSubject
      .pipe(
        skip(1)
      )
      .subscribe((filtersOrderConfiguration: FiltersOrderConfiguration) => {
        expect(filtersOrderConfiguration).toMatchSnapshot();
        done();
      });

    service.updateFilterOrder({
      price: 1,
      difficultLevel: 0,
      language: 2,
      rating: 1,
      duration: 3
    });
  });
});
