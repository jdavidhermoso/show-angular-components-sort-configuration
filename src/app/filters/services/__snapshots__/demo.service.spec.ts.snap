// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersService Should notify when a new value is emitted 1`] = `
Object {
  "difficultLevel": 0,
  "duration": 3,
  "language": 2,
  "price": 1,
  "rating": 1,
}
`;
